js/components/grid/calendar.js.jsx

```diff
@@ -1,301 +1,285 @@
-          lectures={springLectures}
-        />
-      </>
-    )
-  }
+export const Row = props => {
+  // Create a list of lecture objects
+  let lectures = props.lectureSections.map((lectureSection, index, lectureSections) =>
+    createNewLectures(lectureSection, index, lectureSections)
+  )
+  lectures = [].concat.apply([], lectures)
+
+  const fallLectures = lectures.filter(lecture => {
+    return lecture.session === "F" || lecture.session === "Y"
+  })
+
+  const springLectures = lectures.filter(lecture => {
+    return lecture.session === "S" || lecture.session === "Y"
+  })
+
+  // Organize the structure of the <fallSession> and <springSession> 2-D dictionaries
+  let fallSession, springSession
+  ;[fallSession, springSession] = initializeSessions(lectures)
+
+  // For each session, set the 'width' attribute of each Lecture
+  // (go to Lecture constructor for definition of 'width')
+  setWidths(fallSession)
+  setWidths(springSession)
+
+  // Fill session colSpans dictionaries
+  const fallColSpans = { M: 0, T: 0, W: 0, R: 0, F: 0 }
+  const springColSpans = { M: 0, T: 0, W: 0, R: 0, F: 0 }
+  storeColSpans(fallSession, fallColSpans)
+  storeColSpans(springSession, springColSpans)
+
+  // Generate a container for each of the Fall and Spring timetables individually
+  return (
+    <>
+      <TimetableContainer
+        session="F"
+        lecturesByTime={fallSession}
+        headColSpans={fallColSpans}
+        lectures={fallLectures}
+      />
+      <TimetableContainer
+        session="S"
+        lecturesByTime={springSession}
+        headColSpans={springColSpans}
+        lectures={springLectures}
+      />
+    </>
+  )
 }
 
 /*
  * The container specifies formatting for all of the elements wrapped inside,
  * (for example, every element inside a container will follow the same margin rules)
  */
-class TimetableContainer extends React.Component {
-  render() {
-    return (
-      <div className="col-md-5 col-12 timetable-container">
-        <Timetable
-          session={this.props.session}
-          lecturesByTime={this.props.lecturesByTime}
-          headColSpans={this.props.headColSpans}
-          lectures={this.props.lectures}
-        />
-      </div>
-    )
-  }
+const TimetableContainer = props => {
+  return (
+    <div className="col-md-5 col-12 timetable-container">
+      <Timetable
+        session={props.session}
+        lecturesByTime={props.lecturesByTime}
+        headColSpans={props.headColSpans}
+        lectures={props.lectures}
+      />
+    </div>
+  )
 }
 
 /*
  * A <table> element for the specified session
  */
-class Timetable extends React.Component {
-  constructor(props) {
-    super(props)
-    this.modal = React.createRef()
-    this.displayMap = this.displayMap.bind(this)
-  }
-
-  displayMap(session) {
-    this.modal.current.openModal()
-  }
+const Timetable = props => {
+  const modal = createRef()
+  const displayMap = session =>
+    useCallback(() => {
```

Comment:
The documentation for `useCallback` indicates that it should be defined at the top-level of the component. This suggests you should define a first constant `const openMapModal = useCallback(...)` and then use that to define `displayMap`.

Reply:
Should I keep the `session` variable in there, which is not used?